### PR TITLE
Development pool objects and vesting withdrawals algorithms

### PR DESCRIPTION
**Pull request by Andrei-Masilevich**

### CODE REVIEW

File: libraries/chain/services/withdraw_vesting.cpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    const auto& idx = db_impl().get_index<withdraw_vesting_index>().indices().get<by_next_vesting_withdrawal>();
+    auto it = idx.cbegin();
+    const auto it_end = idx.cend();
+    while (it != it_end && it->next_vesting_withdrawal <= until)
+    {
+        ret.push_back(std::cref(*it));
+        ++it;
+    }
+
+    return ret;
```
Comment (abolinbsu) [inline libraries/chain/services/withdraw_vesting.cpp]: use std::move(ret) for all refs_type returnings

Comment (Andrei-Masilevich) [reply libraries/chain/services/withdraw_vesting.cpp]: All return values are either already moved or optimized out, so there is no need to explicitly move with return values.

Comment (Andrei-Masilevich) [reply libraries/chain/services/withdraw_vesting.cpp]: Look: C++11 return value optimization

File: libraries/chain/database/block_tasks/withdrawable_actors_impl.cpp [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+}
+
+void withdrawable_actors_impl::give_scr(const withdrawable_id_type& from,
+                                        const withdrawable_id_type& to,
+                                        const asset& amount)
+{
+    FC_ASSERT(amount.symbol() == SCORUM_SYMBOL);
+    if (amount.amount > 0)
+    {
+        to.visit(give_scr_visitor(_account_service, _dev_pool_service, _dprops_service, _ctx, from, amount));
```
Comment (abolinbsu) [inline libraries/chain/database/block_tasks/withdrawable_actors_impl.cpp]: It's will be more clear if you start with from parameter

**Pull request merged**